packages/task/src/common/task-protocol.ts

```diff
@@ -99,9 +109,106 @@ export interface TaskExitedEvent {
     readonly signal?: string;
 }
 
+export interface TaskOutputEvent {
+    readonly taskId: number;
+    readonly ctx?: string;
+    readonly line: string;
+}
+
+export interface TaskOutputProcessedEvent {
+    readonly taskId: number;
+    readonly ctx?: string;
+    readonly problems?: ProblemMatch[];
+}
+
 export interface TaskClient {
     onTaskExit(event: TaskExitedEvent): void;
     onTaskCreated(event: TaskInfo): void;
     onDidStartTaskProcess(event: TaskInfo): void;
     onDidEndTaskProcess(event: TaskExitedEvent): void;
+    onDidProcessTaskOutput(event: TaskOutputProcessedEvent): void;
+}
+
+export interface TaskDefinition {
+    taskType: string;
+    properties: {
+        required: string[];
+        all: string[];
+    }
+}
+
+export interface TaskDefinitionContribution {
+    type: string;
+    required: string[];
+    properties: {
+        [name: string]: {
+            type: string;
+            description?: string;
+            // tslint:disable-next-line:no-any
+            [additionalProperty: string]: any;
+        }
+    }
+}
+
+export interface WatchingMatcherContribution {
+    // If set to true the background monitor is in active mode when the task starts.
+    // This is equals of issuing a line that matches the beginPattern
+    activeOnStart?: boolean;
+    beginsPattern: string | WatchingPattern;
+    endsPattern: string | WatchingPattern;
+}
+
+export interface ProblemMatcherContribution {
```

Comment:
Some properties are never used, please analyse types and create an issue to implement proper support of them, e.g. here `name` and `filePrefix` for example, i did not have a look at other properties and types.

Reply:
or remove used properties and types

Reply:
Thank you for pointing it out. You are right. `filePrefix` is not used. 
`name` is used as the key in the registration.